test/parallel/test-fs-promises.js

```diff
@@ -57,7 +57,8 @@ assert.strictEqual(
     {
       code: 'ENOENT',
       name: 'Error',
-      message: /^ENOENT: no such file or directory, access/
+      message: /^ENOENT: no such file or directory, access/,
+      stack: /at async Function\.rejects/
```

Comment:
Can you add a test with a closure too?

Reply:
I'm not sure if I understand correctly but I added a test here.
https://github.com/nodejs/node/pull/49849/files#diff-9f28bba0b9948a16a3905ddc942f5c0a49cfd2e34214dc7ce331c7e33fb1c13fR98-R106
To make this work, I had to add `await` before `return` for zero-cost async stack traces (http://bit.ly/v8-zero-cost-async-stack-traces).
But the linter doesn't allow to do `return await` by the [`no-return-await` rule](https://eslint.org/docs/latest/rules/no-return-await).

Reply:
Does the 4-year-old document you link still apply on modern V8 versions? AFAIK the lint rule is there because of assumptions, we should challenge those assumptions and reassess the linter rule if necessary.

Reply:
> Does the 4-year-old document you link still apply on modern V8 versions?

I'm not sure but if I remove `await` the test I added won't pass at least. FWIW the document is linked from https://v8.dev/docs/stack-trace-api#async-stack-traces.

> AFAIK the lint rule is there because of assumptions, we should challenge those assumptions and reassess the linter rule if necessary.

[The documentation](https://eslint.org/docs/latest/rules/no-return-await) says the rule is deprecated.

> This rule was deprecated in ESLint v8.46.0 with no replacement. The original intent of this rule no longer applies due to the fact JavaScript now handles native Promises differently. It can now be slower to remove await rather than keeping it. More technical information can be found in [this V8 blog entry](https://v8.dev/blog/fast-async).

Reply:
If you open a PR to remove that rule, it will probably get accepted.